graphql-dgs-codegen-shared-core/src/main/kotlin/com/netflix/graphql/dgs/client/codegen/GraphQLQueryRequest.kt

```diff
@@ -23,14 +23,18 @@ import graphql.language.OperationDefinition
 import graphql.language.SelectionSet
 import graphql.schema.Coercing
 
-class GraphQLQueryRequest(
+class GraphQLQueryRequest @JvmOverloads constructor(
     val query: GraphQLQuery,
-    val projection: BaseProjectionNode?,
-    scalars: Map<Class<*>, Coercing<*, *>>?
+    val projection: BaseProjectionNode? = null,
+    scalars: Map<Class<*>, Coercing<*, *>>? = null
 ) {
 
-    constructor(query: GraphQLQuery) : this(query, null, null)
-    constructor(query: GraphQLQuery, projection: BaseProjectionNode?) : this(query, projection, null)
+    private var selectionSet: SelectionSet? = null
+
+    @JvmOverloads constructor(query: GraphQLQuery, selectionSet: SelectionSet, scalars: Map<Class<*>, Coercing<*, *>>? = null) : this(query = query, scalars = scalars) {
```

Comment:
Does this work for java? Would be good to double check that the constructor overloads work for creating a GraphQLQueryRequest in java.

Reply:
I had first tried without the overloads that java example tests were failing . It started passing when I added the overloads. Do you want me to test anything specific?

Reply:
@srinivasankavitha

Reply:
Just mainly that it doesn't break for java apps. If you tried it already, that's great. I'll test out your changes  in the next day or so and  post an update.

Reply:
@srinivasankavitha Awesome. Thank you so much

Reply:
Thanks, yes, looks good to me!

Reply:
@srinivasankavitha Thank you. Do you know when it will be released?